cpp/src/arrow/util/value_parsing.h

```diff
@@ -273,6 +273,96 @@ inline bool ParseUnsigned(const char* s, size_t length, uint64_t* out) {
 #undef PARSE_UNSIGNED_ITERATION
 #undef PARSE_UNSIGNED_ITERATION_LAST
 
+#define PARSE_HEX_ITERATION(C_TYPE)                                     \
+  if (length > 0) {                                                     \
```

Comment:
I suggest to make this a function. The use of macros is undesired in the general case.

Reply:
I agree with this recommendation. I was following the pattern of how `ParseUnsigned` and `PARSE_UNSIGNED_ITERATION` is done in the same file. Since I am new to the codebase I decided to follow the patterns I was seeing.